components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/db/TaskQueryHelper.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.ntask.coordination.task.db;
+
+import org.wso2.micro.integrator.ntask.coordination.task.CoordinatedTask;
+
+/**
+ * The class which contains all the data base queries for the task database.
+ */
+public class TaskQueryHelper {
+
+    //task table Name
+    private static final String TABLE_NAME = "COORDINATED_TASK_TABLE";
+
+    // Task table columns
+    public static final String TASK_NAME = "TASK_NAME";
+    public static final String DESTINED_NODE_ID = "DESTINED_NODE_ID";
+    public static final String TASK_STATE = "TASK_STATE";
+
+    public static final String ADD_TASK_IF_NOT_EXISTS =
+            "INSERT IGNORE INTO " + TABLE_NAME + " ( " + TASK_NAME + ", " + DESTINED_NODE_ID + ", " + TASK_STATE + ") "
```

Comment:
Does this work in mssql or other RDBMSs? Let's try to stick to the basic syntaxes so that it won't cause any issues when we add other DB types. WDYT?

Reply:
mssql have some other syntax. but there are should be equivalent syntax. we can fix them while testing (  set constants while initialization the db ), else we won't be able to do batch updates and would have to do one by one. Also we would have to catch data integrity violation each time and process that also. Any other better solution ?

Reply:
Don't we need to capture that integrity violation and may be have a debug log and process each one? I mean this should be called when each task is deployed, isn't it? So does the batch update really contain a batch?

Reply:
we can figure out from result set whether it is udpated or not. We will need batch updates when adding valid tasks which were missed during db connection lost time.

Reply:
which were missed in the sense, deployed while the connection was lost?

Reply:
yes

Reply:
This a 1% situation right? I feel like it's more risky to ignore than it is costly to catch it and log it. Basically IGNORE doesn't only ignore integrity violations it also ignores other types so I don't want it to be hidden. It's your call, 2 sides of the coin.

Reply:
Fixed in https://github.com/wso2/micro-integrator/pull/1227/commits/922fa59ce6aa0e6b2ba7c9f1e90041a79628e2be